File: dev/test_data/golang/test_main/aas_core_meta.v3/input/snippets/Verification/ID_shorts_are_unique.go
// Check that all [aastypes.IReferable.IDShort] are unique among
// `referables`.
func IDShortsAreUnique[R aastypes.IReferable](
	referables []R) bool {
	idShortSet := make(map[string]struct{})

	for _, referable := range referables {
		idShort := referable.IDShort()

		if idShort == nil {
			continue
		}

		_, has := idShortSet[*idShort]
		if has {
			return false
		}

		idShortSet[*idShort] = struct{}{}
	}
	return true
}

File: dev/test_data/test_main/typescript/aas_core_meta.v3/input/snippets/Verification/ID_shorts_are_unique.ts
/**
 * Check that all {@link types.IReferable.idShort}'s are unique
 * among `referables`.
 *
 * @param referables - to be verified
 * @returns `true` if the check passes
 */
export function idShortsAreUnique(
  referables: Iterable<AasTypes.IReferable>
): boolean {
  const idShortSet = new Set<string>();
  for (const referable of referables) {
    if (referable.idShort !== null) {
      if (idShortSet.has(referable.idShort)) {
        return false;
      }

      idShortSet.add(referable.idShort);
    }
  }

  return true;
}
